glusterd2-test.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+# NOTE:
+# This script assumes that glusterfs source (with relevant gerrit change) has
+# been built and/or installed.
+
+# echo commands
+set -x
+
+# if anything fails, we'll abort
+set -e
+
+# install Go version
+REQ_GO_VERSION='1.9.4'
+if ! yum -y install "golang >= $REQ_GO_VERSION"
+then
+        # not the right version, install manually
+        # download URL comes from https://golang.org/dl/
+        curl -O https://storage.googleapis.com/golang/go${REQ_GO_VERSION}.linux-amd64.tar.gz
+        tar xzf go${REQ_GO_VERSION}.linux-amd64.tar.gz -C /usr/local
+        export PATH=$PATH:/usr/local/go/bin
+fi
+
+# also needs git, hg, bzr, svn gcc and make
+# some of these will be used by vendored dependencies
+yum -y install git mercurial bzr subversion gcc make
+
+# set up GOPATH
+mkdir -p go/{pkg,src,bin}
+export GOPATH=$PWD/go
+export PATH=$PATH:$GOPATH/bin
+
+# clone glusterd2 source into GOPATH
+export GD2GIT=https://github.com/gluster/glusterd2.git
+export GD2SRC=$GOPATH/src/github.com/gluster/glusterd2
+mkdir -p $GD2SRC
+git clone $GD2GIT $GD2SRC
+cd $GD2SRC
+
+# install the build and test requirements
+./scripts/install-reqs.sh
+
+# install vendored dependencies
+make vendor-install
+
+# build glusterd2 and cli
+make glusterd2
+make glustercli
+make gd2conf
+
+# run unit tests and other static tests (not really needed for glusterfs tests)
+make test TESTOPTIONS=-v
+
+# run functional tests
+make functest
+
+# TODO: This script doesn't do any cleanup. If centos CI VMs are not disposable
```

Comment:
This will not be run on Centos CI. However, it appears you're using PWD for installing everything. Are you installing anything into the system path at all? If not, the PWD is cleaned up on every run.

Reply:
All things go and glusterd2 resides in `PWD`. Nothing is being installed per se. The tests use `/tmp` directory. That is not cleaned up.

However, glusterfs stuffs that come in from source installation isn't cleaned up by this script. I'm assuming that's done by the same script/entity that installs glusterfs ?

Reply:
Yeah, that gets installed into /build, which we clean at the start of every build. So you're good in that regard.

Reply:
Done :+1: Refreshed the PR accordingly.